packages/server/src/course/course.controller.ts

```diff
@@ -296,4 +307,72 @@ export class CourseController {
     });
     await UserCourseModel.remove(userCourse);
   }
+
+  @Get(':id/ta_check_in_times')
+  @Roles(Role.PROFESSOR)
+  async taCheckinTimes(
+    @Param('id') courseId: number,
+    @Query('startDate') startDate: string,
+    @Query('endDate') endDate: string,
+  ): Promise<TACheckinTimesResponse> {
+    const taEvents = await EventModel.find({
+      where: {
+        eventType: In([
+          EventType.TA_CHECKED_IN,
+          EventType.TA_CHECKED_OUT,
+          EventType.TA_CHECKED_OUT_FORCED,
+        ]),
+        time: Between(startDate, endDate),
+        courseId,
+      },
+      relations: ['user'],
+    });
```

Comment:
Could some of the code in here be moved to a service? I think it would make things more readable plus easier to test and debug

Reply:
agree, I'll port it over later